File: src/app/Models/Announcement.php
<?php

namespace App\Models;

use App\Traits\UsesTenantConnection;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

use App\Traits\Auditable;

class Announcement extends Model
{
    use HasFactory, UsesTenantConnection, Auditable;

    protected $fillable = [
        'title',
        'content',
        'excerpt',
        'cover_image_url',
        'target_audience', // enum: all, teachers, students
        'is_pinned',
        'type',
        'is_published',
        'user_id',
    ];

    protected $casts = [
        'is_published' => 'boolean',
        'is_pinned' => 'boolean',
    ];

    public function author(): BelongsTo
    {
        return $this->belongsTo(User::class, 'user_id');
    }
}

File: floz_mobile/lib/features/student/announcements/domain/entities/announcement.dart
class AnnouncementSummary {
  final int id;
  final String title;
  final String excerpt;
  final String type;
  final bool isPinned;
  final String? coverImageUrl;
  final DateTime? createdAt;

  const AnnouncementSummary({
    required this.id,
    required this.title,
    required this.excerpt,
    required this.type,
    required this.isPinned,
    this.coverImageUrl,
    this.createdAt,
  });
}

class AnnouncementDetail {
  final int id;
  final String title;
  final String content;
  final String type;
  final bool isPinned;
  final String? coverImageUrl;
  final DateTime? createdAt;

  const AnnouncementDetail({
    required this.id,
    required this.title,
    required this.content,
    required this.type,
    required this.isPinned,
    this.coverImageUrl,
    this.createdAt,
  });
}
